osu.Game.Resources/Shaders/sh_TriangleBorder.fs

```diff
@@ -36,7 +36,7 @@ void main(void)
     highp float dst3 = dstToLine(vec2(0.0, 1.0), vec2(0.5, 0.0), pixelPos);
     highp float dst = min(min(dst1, dst2), dst3);
 
-    lowp float alpha = dst < texelSize ? dst / texelSize : smoothstep(texelSize, 0.0, dst - thickness);
+    lowp float alpha = texelSize == 0.0 ? (1.0 - step(thickness, dst)) : (dst < texelSize ? smoothstep(0.0, texelSize, dst) : smoothstep(texelSize, 0.0, dst - thickness));
```

Comment:
Is it safe to say that when `texelSize` is zero, nothing is being displayed to the user? if so, would it not be simpler to just use a constant `alpha` of `1`?

Reply:
In my head when texelSize is 0 it's like having a display of infinite dpi, so no smoothing should be applied.

Reply:
I'd agree that zero texel size turning off smoothing makes sense. So triangles should show, just without smoothing.

From a code quality perspective though, this line is pretty awful with nested ternaries. I'd probably use actual `if`s here, at least on the outer level. As far as the internet tells me ternaries are normal branches in glsl anyway (as in, not vectorised), so there should be no associated performance hit.